src/HealthChecks.UI/Core/HostedService/HealthCheckReportCollector.cs

```diff
@@ -89,6 +96,26 @@ private async Task<UIHealthReport> GetHealthReport(HealthCheckConfiguration conf
                 return UIHealthReport.CreateFrom(exception);
             }
         }
+
+        private Uri GetEndpointUri(HealthCheckConfiguration configuration)
+        {
+             if (endpointAddresses.ContainsKey(configuration.Id))

```

Comment:
As healthchecks trigger far more often than webhooks, we hold parsed absolute uris in an in memory dictionary to avoid parsing of all the endpoints that have already been triggered